app/src/main/java/com/github/kpdandroid/utils/PreferencesManager.kt

```diff
@@ -3,38 +3,59 @@ package com.github.kpdandroid.utils
 import android.content.Context
 import android.util.Log
 import android.util.Size
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.core.content.edit
+import com.github.kpdandroid.utils.detection.DetectionAlgo
 
 private const val TAG = "PreferencesManager"
 
-private const val PREFERENCES_FILE_NAME = "ALGORITHM_PREFERENCES"
+private const val PREFERENCES_FILE_NAME = "preferences"
 
-private const val ALGORITHM_KEY = "algorithm"
-private val ALGORITHM_DEFAULT = KeypointDetectionAlgorithm.NONE.formattedName
+private const val FILE_ALGO_KEY = "file_algorithm"
+private const val CAMERA_ALGO_KEY = "camera_algorithm"
+private val ALGO_DEFAULT = DetectionAlgo.NONE.title
 
 private const val RESOLUTION_KEY = "resolution"
 private val RESOLUTION_DEFAULT = null
 
 class PreferencesManager(context: Context) {
-    private val preferences = context.getSharedPreferences(
+    private val prefs = context.getSharedPreferences(
         PREFERENCES_FILE_NAME,
         Context.MODE_PRIVATE
     )
 
-    var selectedAlgorithmName: String
-        get() = preferences.getString(ALGORITHM_KEY, ALGORITHM_DEFAULT) ?: ALGORITHM_DEFAULT
+    private var fileAlgoNameState by mutableStateOf(
+        prefs.getString(FILE_ALGO_KEY, ALGO_DEFAULT) ?: ALGO_DEFAULT
+    )
+    var fileAlgoName: String
+        get() = fileAlgoNameState
+        set(value) {
+            prefs.edit { putString(FILE_ALGO_KEY, value) }
+            fileAlgoNameState = value
+            Log.i(TAG, "Algorithm $value has been selected for file analysis.")
+        }
+
+    private var cameraAlgoNameState by mutableStateOf(
+        prefs.getString(CAMERA_ALGO_KEY, ALGO_DEFAULT) ?: ALGO_DEFAULT
+    )
+    var cameraAlgoName: String
+        get() = cameraAlgoNameState
         set(value) {
-            preferences.edit { putString(ALGORITHM_KEY, value) }
-            Log.i(TAG, "Algorithm $value has been selected.")
+            prefs.edit { putString(CAMERA_ALGO_KEY, value) }
+            cameraAlgoNameState = value
+            Log.i(TAG, "Algorithm $value has been selected for camera analysis.")
         }
 
-    var selectedResolution: Size?
-        get() = preferences.getString(RESOLUTION_KEY, RESOLUTION_DEFAULT)
-            ?.let { Size.parseSize(it) }
+    private var resolutionState by mutableStateOf(
```

Comment:
I wrote this code precisely because I have no idea how to make it better :)

I would like to replace `State`s with `Flow`s, so that `PreferencesManager` wouldn't be so tied to UI, but it'll only create more code.

Reply:
Ok, I get it.